tests/sanity/extra/licenses.py

```diff
@@ -21,30 +21,35 @@ def find_licenses(filename, relax=False):
     other_license_identifiers = []
     has_copyright = False
-    with open(filename, 'r', encoding='utf-8') as f:
-        for line in f:
-            line = line.rstrip()
-            if 'Copyright ' in line:
-                has_copyright = True
-            if 'Copyright: ' in line:
-                print('%s: found copyright line with "Copyright:". Please remove the colon.' % (filename, ))
-            idx = line.find('SPDX-License-Identifier: ')
-            if idx >= 0:
-                lic_id = line[idx + len('SPDX-License-Identifier: '):]
-                spdx_license_identifiers.extend(lic_id.split(' OR '))
-            if 'GNU General Public License' in line:
-                if 'v3.0+' in line:
-                    other_license_identifiers.append('GPL-3.0-or-later')
-                if 'version 3 or later' in line:
-                    other_license_identifiers.append('GPL-3.0-or-later')
-            if 'Simplified BSD License' in line:
-                other_license_identifiers.append('BSD-2-Clause')
-            if 'Apache License 2.0' in line:
-                other_license_identifiers.append('Apache-2.0')
-            if 'PSF License' in line or 'Python-2.0' in line:
-                other_license_identifiers.append('PSF-2.0')
-            if 'MIT License' in line:
-                other_license_identifiers.append('MIT')
+    try:
+        with open(filename, 'r', encoding='utf-8') as f:
+            for line in f:
+                line = line.rstrip()
+                if 'Copyright ' in line:
+                    has_copyright = True
+                if 'Copyright: ' in line:
+                    print('%s: found copyright line with "Copyright:". Please remove the colon.' % (filename, ))
+                if 'SPDX-FileCopyrightText: ' in line:
+                    has_copyright = True
+                idx = line.find('SPDX-License-Identifier: ')
+                if idx >= 0:
+                    lic_id = line[idx + len('SPDX-License-Identifier: '):]
+                    spdx_license_identifiers.extend(lic_id.split(' OR '))
+                if 'GNU General Public License' in line:
+                    if 'v3.0+' in line:
+                        other_license_identifiers.append('GPL-3.0-or-later')
+                    if 'version 3 or later' in line:
+                        other_license_identifiers.append('GPL-3.0-or-later')
+                if 'Simplified BSD License' in line:
+                    other_license_identifiers.append('BSD-2-Clause')
+                if 'Apache License 2.0' in line:
+                    other_license_identifiers.append('Apache-2.0')
+                if 'PSF License' in line or 'Python-2.0' in line:
+                    other_license_identifiers.append('PSF-2.0')
+                if 'MIT License' in line:
+                    other_license_identifiers.append('MIT')
+    except Exception as exc:
+        print('%s: error while processing file: %s' % (filename, exc))
```

Comment:
So the only change here is putting it in a try-except block?

Reply:
And support for `SPDX-FileCopyrightText:`. The try/except block comes from community.general, the `SPDX-FileCopyrightText:` block change too I think, and the other projects got the `OR` change from this one IIRC. Basically making sure that all projects have essentially the same copy of this whole file. (The only exception is community.dns, which has a special clause for `plugins/public_suffix_list.dat` having a non-GPL license outside of module_utils and modules.)